src/status_im/browser/core.cljs

```diff
@@ -80,7 +80,9 @@
 
 (defn check-if-phishing-url [{:keys [history history-index] :as browser}]
   (let [history-host (http/url-host (try (nth history history-index) (catch js/Error _)))]
-    (assoc browser :unsafe? (dependencies/phishing-detect history-host))))
+    (if history-host
+      (assoc browser :unsafe? (dependencies/phishing-detect history-host))
+      (assoc browser :web-view-error? true))))
```

Comment:
because we don't use `:web-view-error?` anywhere, and there is no spec for it, better to return just browser object

```
(cond-> browser
 (when history-host (assoc :unsafe? (dependencies/phishing-detect history-host)))
```

Reply:
![](https://user-images.githubusercontent.com/7532782/46868322-3c457a00-ce28-11e8-8ef9-739b6ae7b772.png)

This image that was mentioned in the issue seems to be displayed when `:web-view-error?` is true, `unsafe?` would display the phishing warning.

Reply:
i can't find the usage of `:web-view-error?` in the codebase, could you point where :web-view-error? is used

Reply:
i just suggest to do not assign :web-view-error?, in my code example unsafe will not call if history-host is null

Reply:
If you check [this link](https://github.com/status-im/status-react/blob/develop/test/appium/tests/atomic/dapps_and_browsing/test_browsing.py#L23), `Unable to load page` is supposed to be the error displayed for invalid URLs. And [here](https://github.com/status-im/status-react/blob/develop/translations/en.json#L590) is where that message is defined. 

Also what ends up happening with your example is that `history-host` is undefined on first browser load, because `history` is empty at that point. So it doesn't return the browser and that crashes the page. 

But let me know if I got you wrong, I'm very new to Clojure but I'll get things sorted. 

Do you want me to move the invalid URLs check to another function?

Reply:
to avoid null error we just need to do not invoke `phishing-detect`, and also we don't know if it's safe or not, so better to do nothing, something like

```
(if history-host
 (assoc browser :unsafe? (dependencies/phishing-detect history-host)) 
  browser))))
```

but a better way to write this code is 

```
(cond-> browser
 history-host (assoc :unsafe? (dependencies/phishing-detect history-host))
```
thanks for your effort btw ;) clojure ftw :)

Reply:
Got it, thanks. I've pushed a commit to fix this issue.

Reply:
thanks!

Reply:
Seems like I forgot to sign the commit. Should I do a force push?

Reply:
Do you want me to squash my commits in this PR?

Reply:
@dhamaniasad Please yes :)

Reply:
@jeluard Done.